Content.Client/Atmos/Overlays/AtmosDebugGraphOverlay.cs

```diff
@@ -0,0 +1,241 @@
+﻿using Content.Client.Atmos.EntitySystems;
+using Content.Client.Resources;
+using Content.Shared.Atmos;
+using Content.Shared.Atmos.EntitySystems;
+using Robust.Client.Graphics;
+using Robust.Client.ResourceManagement;
+using Robust.Shared.Enums;
+using Robust.Shared.Map;
+
+namespace Content.Client.Atmos.Overlays;
+
+public sealed class AtmosDebugGraphOverlay : Overlay
+{
+    private readonly AtmosDebugOverlaySystem _atmosDebugOverlaySystem;
+
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly IEyeManager _eyeManager = default!;
+    [Dependency] private readonly IResourceCache _resourceCache = default!;
+
+    private readonly Font _font;
+    private static readonly Color BlockedDirectionColor = new(96, 32, 32);
+
+    public override OverlaySpace Space => OverlaySpace.WorldSpace | OverlaySpace.ScreenSpace;
+
+    public AtmosDebugGraphOverlay()
+    {
+        IoCManager.InjectDependencies(this);
+
+        _atmosDebugOverlaySystem = EntitySystem.Get<AtmosDebugOverlaySystem>();
+        _font = _resourceCache.GetFont("/Fonts/NotoSans/NotoSans-Regular.ttf", 10);
+    }
+
+    protected override void Draw(in OverlayDrawArgs args)
+    {
+        if ((args.Space & OverlaySpace.WorldSpace) != 0)
+        {
+            WorldSpaceDraw(args);
+        }
+        else if ((args.Space & OverlaySpace.ScreenSpace) != 0)
+        {
+            ScreenSpaceDraw(args);
+        }
+    }
+
+    private void WorldSpaceDraw(in OverlayDrawArgs args)
+    {
+        var drawHandle = args.WorldHandle;
+
+        var mapId = args.Viewport.Eye!.Position.MapId;
+        var worldBounds = args.WorldBounds;
+
+        foreach (var mapGrid in _mapManager.FindGridsIntersecting(mapId, worldBounds))
+        {
+            if (!_atmosDebugOverlaySystem.HasData(mapGrid.Index))
+                continue;
+
+            drawHandle.SetTransform(mapGrid.WorldMatrix);
+
+            foreach (var tile in mapGrid.GetTilesIntersecting(worldBounds))
+            {
+                var dataMaybeNull = _atmosDebugOverlaySystem.GetData(mapGrid.Index, tile.GridIndices);
+                if (!dataMaybeNull.HasValue) continue;
+
+                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.BlockDirections) != 0)
+                    DrawBlockDirections(drawHandle, tile, dataMaybeNull.Value);
+                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.GasMoles) != 0)
+                    DrawGasMolesBarChart(drawHandle, tile, dataMaybeNull.Value);
+                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.FlowDirections) != 0)
+                    DrawFlowDirections(drawHandle, tile, dataMaybeNull.Value);
```

Comment:
```suggestion

                if (_atmosDebugOverlaySystem.GetData(mapGrid.Index, tile.GridIndices) is not { } data) continue;

                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.BlockDirections) != 0)
                    DrawBlockDirections(drawHandle, tile, data);
                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.GasMoles) != 0)
                    DrawGasMolesBarChart(drawHandle, tile, data);
                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.FlowDirections) != 0)
                    DrawFlowDirections(drawHandle, tile, data);
```

believe this works to make it a bit cleaner